### ISSUE TITLE
synapse fails to start when using LetsEncrypt certs

### ISSUE THREAD

**Issue by swk** [matrix-org/synapse#2911]
### Description

Debian 8, Certbot (from their git), installing Synapse from the matrix provided debian repositories
and following the Matrix Synapse/LetsEncrypt configuration guide available at https://matrix.org/docs/guides/lets-encrypt.html fails to produced a working system.

"journal -xn" reveals that matrix is attempting to read the files, but it cant not due to perms errors.
- matrix python[3480]: OSError: [Errno 13] Permission denied:'/etc/letsencrypt/live/${myhostdom}/privkey.pem'

This is caused by Synapse starting as a non-root user as the files in /etc/letsencrypt are perms 600 and not readable by the matrix-synapse user.

Work around is to copy the files to another dir  where they can be chown'd to the matrix-synapse user while keeping them chmod'd to 600. 

This will break auto-updating of the certificate material with certbot and will require additional coding to work around.

In an ideal world, synapse would start with the correct permissions to read things like this, then afterwards drop permissions that were no longer required. (this is like how apache, nginx etc start up as root so they can read configs, bind to ports <1024 etc, then drop permissions to their normal service user as to not expose the system to hack attempts).

### Version information

- **Homeserver**: Was this issue identified on matrix.org or another homeserver?
- **Version**:   Synapse 0.26.0-1
- **Install method**: package install from matrix.org repos
- **Platform**:  Debian8, KVM, 4 cores, 4096G ram, static public IP

**Comment by Half-Shot** [comment]
It's the recommendation of the community at least, if not officially to use the self signed certificates that Synapse generates for federation traffic, and use your LE certs for client traffic with Nginx/Apache

**Comment by swk** [comment]
either way this is not expected behavior as per the documentation

**Comment by Tealk** [comment]
have the same Problem
`Mär 31 02:55:39 v22017114352656784 python[26489]: OSError: [Errno 13] Permission denied: '/etc/letsencrypt/live/xxx/privkey.pem'
Mär 31 02:55:39 v22017114352656784 systemd[1]: matrix-synapse.service: Control process exited, code=exited status=1
Mär 31 02:55:39 v22017114352656784 systemd[1]: Failed to start Synapse Matrix homeserver.
`

**Comment by swk** [comment]
@tealk setting the perms on the live/mydomain/privkey.pem will not work as this is just a symlink to the actual files created by certbot. Changing the perms on the actual files will cause other issues. It will at a minimum a security risk as it then exposes the private key to any users on the system.

**Comment by Tealk** [comment]
@swk 
you mean in /etc/letsencrypt/archive/xxx
there i have set permissions to 777 too

**Comment by KlausHans** [comment]
Got the same error. Commented here a while ago but this was maybe the wrong place: https://github.com/matrix-org/synapse/issues/2379

As i wrote here, i run in this issue if i want to switch from the source build to the Debian package. My current workaround is to keep my source build active. Just tried it again today.

Strange that this issue is still there and not more people are complaining. It makes it impossible to use the debian package build with letsencrypt if i am not mistaken.

**Comment by richvdh** [comment]
> It makes it impossible to use the debian package build with letsencrypt if i am not mistaken.

You are mistaken. Firstly, as Half-Shot says there really is no need to use a Let's Encrypt cert for federation; secondly you just need to fix the permissions on the file (and have whatever script you use to renew the cert do the same).

This is really nothing to do with Debian packaging, but I guess the reason people find it works with the source build is that they are running the source build as `root`, which of course is a terrible idea but will at least mask their permissions problems.

If you really want to use Let's Encrypt for federation:

> Work around is to copy the files to another dir where they can be chown'd to the matrix-synapse user while keeping them chmod'd to 600.

That's one option. Another would be to run synapse in a group which has read access to `/etc/letsencrypt/...` and chmod the cert to 640. Either of these things can be done with a `--deploy-hook` to certbot.

I've removed the offending guide, which was written by a well-meaning but poorly informed contributor, from the website.

Closing this bug as synapse most certainly does *not* fail to start when using LetsEncrypt certs, provided it has permission to access the file.
